### PR TITLE
JIT the quantum circuit

### LINKED ISSUE #13: Big training speed-up for circuit simulation.
Here are two very simple changes can speed up training **a lot.**

1. The default pennylane device: `dev = qml.device("default.qubit", wires=num_qubits)` may bit `jitted` using JAX very easily. 

2. `default.qubit` may use classical backpropagation for gradient calculations. It's not clear if it's using it by default, so specify it explicitly.

According to https://docs.pennylane.ai/en/stable/introduction/interfaces/jax.html#using-jax-jit-on-qnodes

This works

```python
dev = qml.device('default.qubit', wires=2)

@jax.jit  # QNode calls will now be jitted, and should run faster.
@qml.qnode(dev, interface='jax', diff_method='backprop')
def circuit4(phi, theta):
    qml.RX(phi[0], wires=0)
    qml.RZ(phi[1], wires=1)
    qml.CNOT(wires=[0, 1])
    qml.RX(theta, wires=0)
    return qml.expval(qml.PauliZ(0))

```

We have of course dropped the decorator syntactic sugar, so you can just call `jax.jit(qnode)`

### PR DESCRIPTION
**Pull request by minhtriet**
Fix #13

### CODE REVIEW

**Review by jackbaker1001 — CHANGES_REQUESTED** [review]
You are jitting the Qnode everytime you are calling the qnode. Jit it once and store the jitted qnode as an object.

**Review by jackbaker1001 — APPROVED** [review]
